kedro/framework/cli/catalog.py

```diff
@@ -233,7 +233,10 @@ def resolve_patterns(metadata: ProjectMetadata, env: str) -> None:
     context = session.load_context()
 
     catalog_config = context.config_loader["catalog"]
-    data_catalog = DataCatalog.from_config(catalog_config)
+    credentials_config = context.config_loader.get("credentials", None)
```

Comment:
I'm aware that using `.get()` is inconsistent with how we get the catalog from the config loader, but there's a higher chance a user doesn't have any credentials. I'm also wondering if using `.get()` is problematic with how stuff works in python 3.12 (cc: @AhdraMeraliQB @ankatiyar)

Reply:
I think using `.get()` should be fine with python 3.12 after #3587